index.js

```diff
@@ -2,26 +2,17 @@ var loaderUtils = require("loader-utils");
 
 module.exports = function() {}
 module.exports.pitch = function applyLoader(remainingRequest) {
-  var query = loaderUtils.parseQuery(this.query);
+  var opts = this.resourceQuery ? loaderUtils.parseQuery(this.resourceQuery) : loaderUtils.getOptions(this) || {};
```

Comment:
From what I can see in the [webpack readme](http://webpack.github.io/docs/loaders.html#resource) this line would mean `apply?args[]=1!functionReturningLoader?args[]=2` would call it like `functionReturningLoader(2)` instead of `functionReturningLoader(1)`. Is this really desired? My intuition says this probably isn't expected behavior for the end user.